CDCgov/prime-simplereport
# Johanna/6723 add destination disease columns upload table

# BACKEND PULL REQUEST

## Related Issue

- Resolves #6723 

## Changes Proposed

- Creates the model for the new table `UploadDiseaseDetails` and corresponding repository.
- Updates to `TestResultUploadService` include:
   - CVS validation logic moved to its own method.
   - Processing of the ReportStream response (covid and universal pipeline) have been moved to their own method.
   - Method `saveSubmissionToDb` now receives the destination pipeline and the disease/count breakdown.
- Creates auxiliary classes :
   - `FHIRBundleRecord` : contains the serialized fhir bundle plus the diseases/count breakdown as metadata.
   - `CovidSummisionSummary`: contains the RS response and other information relevant to covid reporting (e.g.: submissionId, org, destination).
   - `UniversalSummisionSummary`: contains the RS response and other information relevant to universal reporting (e.g.: submissionId, org, destination).
   - `Pipeline`: new type to represent COVID | UNIVERSAL pipeline names.
- `TestResultUpload` now uses the @builder annotation.
 

## Testing
Code is available in dev2.

- Upload a file and verify that the upload records and the disease details are correctly logged in our DB.
- Verify validations work as usual.
